hs_core/models.py

```diff
@@ -718,26 +729,25 @@ def update(cls, element_id, **kwargs):
         metadata_obj = kwargs['content_object']
         # get matching resource
         resource = BaseResource.objects.filter(object_id=metadata_obj.id).first()
-
+        publisher_CUAHSI = "Consortium of Universities for the Advancement of Hydrologic Science, Inc. (CUAHSI)"
         if 'name' in kwargs:
             if pub.name.lower() != kwargs['name'].lower():
-                if pub.name.lower() == 'hydroshare':
+                if pub.name.lower() == publisher_CUAHSI.lower():
                     if resource.files.all():
-                        raise ValidationError("Publisher 'HydroShare' can't be changed for a resource that has "
+                        raise ValidationError("Publisher 'HydroShare/CUAHSI' can't be changed for a resource that has "
                                               "content files.")
-                elif kwargs['name'].lower() == 'hydroshare':
+                elif kwargs['name'].lower() == publisher_CUAHSI.lower():
                     if not resource.files.all():
-                        raise ValidationError("'HydroShare' can't be a publisher for a resource that has no "
+                        raise ValidationError("'HydroShare/CUAHSI' can't be a publisher for a resource that has no "
                                               "content files.")
 
-                if resource.files.all():
-                    kwargs['name'] = 'HydroShare'
-
         if 'url' in kwargs:
-            if pub.url != kwargs['url']:
-                # make sure we are not changing the url for hydroshare publisher
-                if pub.name.lower() == 'hydroshare':
-                    kwargs['url'] = 'http://hydroshare.org'
+            if pub.url.lower() != kwargs['url'].lower():
+                # make sure we are not changing the url for hydroshare publisher when there are content files
+                if pub.url.lower() == 'https://www.cuahsi.org':
+                    if resource.files.all():
+                        raise ValidationError("Publisher 'HydroShare/CUAHSI' can't be changed for a resource that has "
+                                              "content files.")
```

Comment:
Same comment as above, no additional check is needed. A simple ValidationError() can be raised indicating Publisher element is not allowed to be updated.